src/graph/serializers/serialize_store.c

```diff
@@ -27,18 +27,16 @@ void* RdbLoadStore(RedisModuleIO *rdb) {
   s->properties = NewTrieMap();
   size_t len = 0;
   uint64_t propCount = RedisModule_LoadUnsigned(rdb);
-  char *propStrings[propCount];
+
   for(int i = 0; i < propCount; i++) {
-    propStrings[i] = RedisModule_LoadStringBuffer(rdb, &len);
-    propStrings[i][len] = '\0';
-  }
+    // Load property string from RDB file
+    char *prop = RedisModule_LoadStringBuffer(rdb, &len);
 
-  LabelStore_UpdateSchema(s, propCount, propStrings);
+    // Add the string directly to the store's triemap using the RDB-given length
+    TrieMap_Add(s->properties, prop, len, NULL, TrieMap_NOP_REPLACE);
```

Comment:
I prefer we don't access store internals (properties), but I don't see a good way to go around it...